ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Credits to https://github.com/darrylhodgins/typescript-memoize
+ */
+
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+
+interface MemoizeArgs {
+  expiring?: number;
+  hashFunction?: boolean | ((...args: any[]) => any);
+  tags?: string[];
+}
+
+export function Memoize(args?: MemoizeArgs | MemoizeArgs["hashFunction"]) {
+  let hashFunction: MemoizeArgs["hashFunction"];
+  let duration: MemoizeArgs["expiring"];
```

Comment:
Let's just use consistent naming, I'd go with duration, expiring doesn't make sense to me.

Reply:
duration renamed